services/wfm/src/main/java/org/openkilda/wfm/share/mappers/FlowMapper.java

```diff
@@ -0,0 +1,177 @@
+    public abstract FlowDto map(Flow flow);
+
+    /**
+     * Builds a {@link Flow} with provided data from {@link FlowDto}.
+     * <p/>
+     * <strong>Be careful as it creates a dummy switch objects for srcSwitch and destSwitch properties
+     * with only switchId filled.</strong>
+     */
+    @Mapping(source = "sourcePort", target = "srcPort")
+    @Mapping(source = "sourceVlan", target = "srcVlan")
+    @Mapping(source = "destinationPort", target = "destPort")
+    @Mapping(source = "destinationVlan", target = "destVlan")
+    @Mapping(target = "srcSwitch",
+            expression = "java(org.openkilda.model.Switch.builder().switchId(flow.getSourceSwitch()).build())")
+    @Mapping(target = "destSwitch",
+            expression = "java(org.openkilda.model.Switch.builder().switchId(flow.getDestinationSwitch()).build())")
+    @Mapping(source = "state", target = "status")
+    @Mapping(source = "lastUpdated", target = "timeModify")
+    public abstract Flow map(FlowDto flow);
+
+    /**
+     * Convert {@link PathNode} to {@link FlowPath.Node}.
+     */
+    public abstract FlowPath.Node map(PathNode p);
+
+    /**
+     * Convert {@link FlowPath.Node } to {@link PathNode}.
+     */
+    public abstract PathNode map(FlowPath.Node p);
+
+    /**
+     * Convert {@link FlowPath} to {@link PathInfoData}.
+     */
+    @Mapping(source = "nodes", target = "path")
+    public abstract PathInfoData map(FlowPath p);
+
+    /**
+     * Convert {@link PathInfoData} to {@link FlowPath}.
+     */
+    @Mapping(source = "path", target = "nodes")
+    public abstract FlowPath map(PathInfoData p);
+
+
+    /**
+     * Convert {@link FlowPair} to {@link FlowPairDto}.
+     */
+    public FlowPairDto<FlowDto, FlowDto> map(FlowPair flowPair) {
+        if (flowPair == null) {
+            return null;
+        }
+
+        return new FlowPairDto<>(map(flowPair.getForward()), map(flowPair.getReverse()));
+    }
+
+    /**
+     * Convert {@link FlowPairDto} to {@link FlowPair}.
+     */
+    public FlowPair map(FlowPairDto<FlowDto, FlowDto> flowPair) {
+        if (flowPair == null) {
+            return null;
+        }
+
+        return FlowPair.builder().forward(map(flowPair.getLeft())).reverse(map(flowPair.getRight())).build();
+    }
+
+    /**
+     * Convert {@link String} to {@link Instant}.
+     */
+    public Instant map(String value) {
+        if (value == null) {
+            return null;
+        }
+
+        return Instant.parse(value);
+    }
+
+    /**
+     * Convert {@link Instant} to {@link String}.
+     */
+    public String map(Instant value) {
+        if (value == null) {
+            return null;
+        }
+
+        return value.toString();
+    }
+
+    /**
+     * Convert {@link FlowStatus} to {@link FlowState}.
+     */
+    public FlowState map(FlowStatus status) {
+        if (status == null) {
+            return null;
+        }
+
+        switch (status) {
+            default:
+            case IN_PROGRESS:
```

Comment:
minor: is it OK to have default state "in progress" for flow?

Reply:
Replaced with specific enum values.